sadx-fixed-edition/mod.cpp

```diff
@@ -126,5 +279,43 @@ extern "C"
 		*reinterpret_cast<NJS_OBJECT*>(0x0279D364) = object_000A0E58;
 
 		PlaySegaSonicTeamVoice_init();
+
+		// Fix Mystic Ruins base
+		if (GetModuleHandle(TEXT("DCMods_Main.dll")) == nullptr) FixMRBase_Apply(path, helperFunctions);
+
+		// Fix Egg Carrier Garden ocean animation
+		((LandTable *)0x3405E54)->Col[74].Flags = 0x84000002;
+		((NJS_MATERIAL*)0x033FE3F8)->diffuse.color = 0x7FB2B2B2;
+
+		// Fix the water ripple created by air bubbles
+		WriteJump(reinterpret_cast<void*>(0x7A81A0), FixedBubbleRipple);
+
+		// Fix Perfect Chaos damage animation
+		if (GetModuleHandle(TEXT("DCMods_Main.dll")) == nullptr)
+		{
+			WriteJump(reinterpret_cast<void*>(0x005632F0), Chaos7Explosion_DisplayX);
+			WriteJump(reinterpret_cast<void*>(0x005633C0), Chaos7Damage_DisplayX);
+		}
+	}
+	EXPORT void __cdecl OnFrame()
+	{
+		if (GetModuleHandle(TEXT("DLCs_Main.dll")) == nullptr)
```

Comment:
This DLL being the exception to the local variable suggestion; it should have a global variable stored instead. This is very inefficient as-is.